app/javascript/bundles/RestaurantsApp/components/AddReview.jsx

```diff
@@ -0,0 +1,114 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Formsy from 'formsy-react';
+import TextInput from '../../Forms/components/TextInput';
+import StarsRatingFilter from "../../StarsRating/components/StarsRatingFilter"
+
+
+export default class AddReviewForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isFormValid: false,
+            isFormSubmitted: false,
+            userRating: 0,
+            isAddingReview: false,
+        }
+        this.submit = this.submit.bind(this);
+        this.setValid = this.setValid.bind(this);
+        this.setInvalid = this.setInvalid.bind(this);
+
+    }
+
+    static propTypes = {
+        restaurant: PropTypes.object.isRequired,
+        onClose: PropTypes.func.isRequired,
+        onAdd: PropTypes.func.isRequired,
+    };
+
+    getInitialState() {
+        return {
+            isFormValid: false
+        }
+    }
+
+    setValid() {
+        this.setState({
+            isFormValid: true
+        });
+    }
+
+    setInvalid() {
+        this.setState({
+            isFormValid: false
+        });
+    }
+
+    submit(model) {
+        this.setState({isFormSubmitted: true});
+        if (this.state.isFormValid) {
+            this.setState({isAddingReview: true});
+            const onClose = this.props.onClose;
+            const onAdd = this.props.onAdd;
+            const review = {
+                review: {
+                    rating: this.state.userRating,
+                    restaurant_id: this.props.restaurant.id,
+                    name: model.userName,
+                    comment: model.comment
+                }
+            };
+
+            const data = JSON.stringify(review);
+            fetch("/reviews",
+                {
+                    method: "POST",
+                    body: data,
+                    headers: {
+                        'Accept': 'application/json, text/plain, */*',
+                        'Content-Type': 'application/json'
+                    },
+                })
+                .then(function (response) {
+                    onClose();
+                    onAdd();
+                })
+                .catch(function (error) {
+                    console.log(error);
+                });
+
+        }
+    }
+
+    render() {
+        let formClass = this.state.isFormSubmitted ? "submitted" : "not-submitted";
+        formClass += " form";
```

Comment:
There's a great library to help with class name handling, that's how Facebook originally did their classname handling (probably still do):
https://github.com/JedWatson/classnames

```js
import cx from 'classnames';

const className = cx('static-class', foo, /*...,*/ {
  bar: value, // Will render if value is truthy.
  // ...
});
```